src/sampler_learning.py

```diff
@@ -29,9 +43,55 @@ def learn_samplers(strips_ops: List[STRIPSOperator],
             raise NotImplementedError("Unknown sampler_learner: "
                                       f"{CFG.sampler_learner}")
         samplers.append(sampler)
+    if sampler_learner == "oracle":
+        assert not gt_nsrts, f"Can't use oracle samplers, {len(gt_nsrts)} " \
+            "oracle operator(s) were not matched to a learned operator!"
     return samplers
 
 
+def _extract_oracle_sampler(op: STRIPSOperator,
+                            param_option: ParameterizedOption,
+                            option_vars: Sequence[Variable],
+                            gt_nsrts: Set[NSRT]) -> NSRTSampler:
+    """Extract the oracle sampler matching the given STRIPSOperator from the
+    ground truth operators defined in approaches/oracle_approach.py.
+
+    If no ground truth operator can be unified with the given one, just
+    returns a random sampler.
+    """
+    matching_nsrt = None
+    matching_sub = None
+    for nsrt in gt_nsrts:
+        suc, sub = utils.unify_preconds_effects_options(
+            frozenset(op.preconditions), frozenset(nsrt.preconditions),
+            frozenset(op.add_effects), frozenset(nsrt.add_effects),
+            frozenset(op.delete_effects), frozenset(nsrt.delete_effects),
+            param_option, nsrt.option, tuple(option_vars),
+            tuple(nsrt.option_vars))
+        if suc:
+            matching_nsrt = nsrt
+            matching_sub = sub
+            break
+    if matching_nsrt is None:
```

Comment:
i'm a little worried that we won't know when this is happening. do you know if this happens in cover / blocks right now? can you make an option so that an error is raised here instead of using a random sampler, and set that option to true for at least cover?